[BUG] test_edge_cut_clustering fails intermittently on dolphin dataset

Saw it on several PRs, various configs. Had to rerun tests a few times.

```

=================================== FAILURES ===================================

_____________ test_edge_cut_clustering[2-../datasets/dolphins.csv] _____________



graph_file = '../datasets/dolphins.csv', partitions = 2



    @pytest.mark.parametrize("graph_file", utils.DATASETS)

    @pytest.mark.parametrize("partitions", PARTITIONS)

    def test_edge_cut_clustering(graph_file, partitions):

        gc.collect()

    

        # Read in the graph and get a cugraph object

        cu_M = utils.read_csv_file(graph_file, read_weights_in_sp=False)

    

        G_edge = cugraph.Graph()

        G_edge.from_cudf_edgelist(cu_M, source="0", destination="1")

    

        # Get the edge_cut score for partitioning versus random assignment

        """cu_vid, cu_score = cugraph_call(G_adj, partitions)

        rand_vid, rand_score = random_call(G_adj, partitions)

        """

        # Assert that the partitioning has better edge_cut than the random

        # assignment

        """assert cu_score < rand_score"""

    

        # Get the edge_cut score for partitioning versus random assignment

        cu_vid, cu_score = cugraph_call(G_edge, partitions)

        rand_vid, rand_score = random_call(G_edge, partitions)

    

        # Assert that the partitioning has better edge_cut than the random

        # assignment

        print(cu_score, rand_score)

>       assert cu_score < rand_score

E       assert 69.0 < 59.0



cugraph/tests/test_balanced_cut.py:77: AssertionError

----------------------------- Captured stdout call -----------------------------

Reading ../datasets/dolphins.csv...

69.0 59.0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -67,6 +67,7 @@
 - PR #1025: Explicitly skip raft test folder for pytest 6.0.0
 - PR #1027 Fix documentation
 - PR #1033 Fix reparition error in big datasets, updated coroutine, fixed warnings
+- PR #1040 Fix spectral clustering renumbering issue
 
 # cuGraph 0.14.0 (03 Jun 2020)
 

--- a/python/cugraph/community/spectral_clustering.py
+++ b/python/cugraph/community/spectral_clustering.py
@@ -82,7 +82,7 @@ def spectralBalancedCutClustering(
     )
 
     if G.renumbered:
-        # FIXME:  This is a hack to get around an 
+        # FIXME:  This is a hack to get around an
         # API problem.  The spectral API assumes that
         # the data frame remains in internal vertex
         # id order.  It should not do that.

--- a/python/cugraph/community/spectral_clustering.py
+++ b/python/cugraph/community/spectral_clustering.py
@@ -82,7 +82,11 @@ def spectralBalancedCutClustering(
     )
 
     if G.renumbered:
-        df = G.unrenumber(df, "vertex")
+        # FIXME:  This is a hack to get around an 
+        # API problem.  The spectral API assumes that
+        # the data frame remains in internal vertex
+        # id order.  It should not do that.
+        df = G.unrenumber(df, "vertex", preserve_order=True)
 
     return df
 

